src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2106,6 +2108,20 @@ private IOperation VisitConditionalExpression(IOperation condition, bool sense,
                 {
                     break;
                 }
+
+                // ProduceIsSense will add a UnaryOperator if sense is false, so don't bother erasing
+                // this one if it would need to add it back in
+                if (unOp.Operand.Kind != OperationKind.UnaryOperator &&
```

Comment:
>if (unOp.Operand.Kind != OperationKind.UnaryOperator && [](start = 16, length = 55)

Consider if the following approach is simpler and involves less logic duplication:
- Remember the last Not that we about to dig through here.
- pass it to the ProduceIsSense
- Change ProduceIsSense to simply update the operand of that logical Not if we have it and need to invert the value.

BTW, it looks like ProduceIsSense can be inlined into this function because there are no more consumers.